### DISCUSSION TITLE
Ссылка в нотификации

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#56] [Как это сделать?] [answered]
Добрый день!
Хочется все нотификации сделать с сылками на сущности, из вызвавшие, например задачи, сделки, заявки и т.д.
Как это сделать?
Каким то образом присвоить url и передавать через htmltext, или вставлять переменный код во временную таблицу, вызываемую linktodatatable?

**Comment by juramebel** [comment]
И еще сразу вопрос. 
Можно ли написать какой то универсальный код, который бы формировал для каждого пользователя что то типа хроники - список всех изменений в таблицах и полях, где он имеет доступ.
Может быть  через logRowList?

**Reply by ttm-alexeypolunin** [reply]
Вот с привязкой к таймкоду: https://youtu.be/E9yeH52zd60?t=339

Сейчас доступ к этой таблице имеют Создатели и пользователи у которых стоит «управление пользователями». Надо немного доработать, я вижу там у нас есть ошибки в каких-то случаях и надо запретить «юзер-админам» в логе видеть Создателей, АПИ и технических и вообще перенести видимость этой таблицы в sudo.

Про пользователей: https://docs.totum.online/user-management

**Reply by juramebel** [reply]
Но это только относительно действий пользователя.
Тут идея немного другая.
Вот я сел на рабочее место и быстро просмотрел все изменения в объектах, где я участвую - но которые делали именно другие пользователи. 
Чтобы быстро обзорно понять, что происходило.

**Reply by ttm-alexeypolunin** [reply]
Я задачу понял, было бы хорошо наверное для циклов иметь такой встроенный функционал — я запишу в очередь на обдумывание.

Сейчас можно его накидать вручную, но это прям морочиться надо через [Код действия таблицы](https://docs.totum.online/tables#table-settings-default_action) и отслеживание переменной 

$#changes — переменная доступная только в коде-действия таблицы. Возвращает информацию об изменившихся полях в таблице в виде row:

+ deleted — id удаленнх строк.
+ restored — id восстановленных строк.
+ added — id добавленных строк.
+ changed — row ключами которого являются id измененных строк, а значениями список из name полей измененных в строке.
+ reorderedIds — id строк с измененным порядком по n.

**Reply by juramebel** [reply]
Использовали похожий функционал в одной crm - он там называется хроника. 
Хотели в тотуме его воспроизвести)
в ней видишь все события, которые тебя касаются в одной ленте. Удобно

**Reply by EKhorikov** [reply]
Если будет результат, и будет не сложно, выложите как реализовали. Функционал действительно полезный.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Нотификация может быть как [linkToDataHtml](https://docs.totum.online/functions#fn-linkToDataHtml) и содержать ссылку например на цикл:

```
= : linkToDataHtml(title: "Link sample"; html: str`"<a href ='" + $link + "'>link</a>"`)

link: str`"https://" + $#nh + "/Table/132/726/" + $cycle_id + "/727"`

	cycle_id: "..."

``` 

Структура путей в Тотум для циклов такая `/Table/TREE_ID/ID_CYCLES/CYCLE_ID/ID_TABLE_IN_CYCLE`:

те `/Table/132/726/30/727` это:

+ 132 — ветка дерева
+ 726 — id таблицы циклов
+ 30 — id цикла
+ 727 — id таблицы внутри цикла


Также нотификация может быть временной таблицей — те на плашке нотификации прям будет временная таблица загружена. Туда можно вставить кнопки с `linkToTable` например: https://docs.totum.online/notification-in-the-form-of-a-time-table
